### ISSUE TITLE
Australian artifact for BC 20.5 OnPrem contains an expired license file

### ISSUE THREAD

**Issue by DavidHrbek** [microsoft/navcontainerhelper#2955]
**Australian artifact for BC 20.5 OnPrem contains an expired license file**
There is an expired license file in the following artifact [https://bcartifacts.azureedge.net/onprem/20.5.45456.45889/au](https://bcartifacts.azureedge.net/onprem/20.5.45456.45889/au)
When we download the mentioned artifact, we can see the license file (C:\bcartifacts.cache\onprem\20.5.45456.45889\au\database\Cronus.flf) has been generated with the expiration date 3/24/2023:

![image](https://user-images.githubusercontent.com/120056000/228550741-a11a61f7-c3da-4f30-bc44-59b43906aa1c.png)

I believe it is not correct. I compared it e.g. with a license from the W1 artifact for the same version - [https://bcartifacts.azureedge.net/onprem/20.5.45456.45889/w1](https://bcartifacts.azureedge.net/onprem/20.5.45456.45889/w1) - and it has a much later expiration date:

![image](https://user-images.githubusercontent.com/120056000/228552674-a33269f0-9e60-4554-a90e-9bad81109e95.png)

When we try to use this artifact for the New-BcContainer or the AL-Go for GitHub CI/CD pipeline, the process fails, because it tries to start the BC instance with this default license for the first time (even if we overwrite the license with our own development license via the -licenseFile parameter).

**Example**
```
$artifactUrl = Get-BCArtifactUrl -type OnPrem -country AU -version "20.5" 
$containerName = "BC205"
New-BcContainer `
    -accept_eula `
    -containerName $containerName `
    -artifactUrl $artifactUrl `
    -auth Windows `
    -updateHosts `
    -verbose `
    -isolation hyperv
```

**Full output of scripts**
```
PS> $artifactUrl = Get-BCArtifactUrl -type OnPrem -country AU -version "20.5" 

PS> $containerName = "BC205"

PS> New-BcContainer `
    -accept_eula `
    -containerName $containerName `
    -artifactUrl $artifactUrl `
    -auth Windows `
    -updateHosts `
    -verbose `
    -isolation hyperv
BcContainerHelper is version 4.0.14
BcContainerHelper is running as administrator
HyperV is Enabled
UsePsSession is True
Host is Microsoft Windows 10 Pro - 10.0.19045.2728
Docker Client Version is 20.10.22
Docker Server Version is 20.10.22
Removing entries from hosts
Removing BC205 from container hosts file
Removing BC205-* from container hosts file
Removing Desktop shortcuts
Removing C:\ProgramData\BcContainerHelper\Extensions\BC205
Fetching all docker images
Fetching all docker volumes
INFO: Windows 10 21H1/21H2 images are not yet available, using 2004 as these are found to work better than 20H2 on 21H1/21H2
Using image mcr.microsoft.com/businesscentral:10.0.19041.1415
Creating Container BC205
Style: onprem
Multitenant: No
Version: 20.5.45456.45889
Platform: 20.0.45448.45882
Generic Tag: 1.0.2.14
Container OS Version: 10.0.19041.1415 (2004)
Host OS Version: 10.0.19045.2728 (22H2)
Using hyperv isolation
Using locale en-AU
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Files in C:\ProgramData\BcContainerHelper\Extensions\BC205\my:
- AdditionalOutput.ps1
- MainLoop.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container BC205 from image mcr.microsoft.com/businesscentral:10.0.19041.1415
3a9827206de5fdce4857b116844a13a003fa33abf64cbe942c22d3b228faa73f
Waiting for container BC205 to be ready
Using artifactUrl https://bcartifacts.azureedge.net/onprem/20.5.45456.45889/au
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\onprem\20.5.45456.45889\platform\ServiceTier\Program Files
c:\dl\onprem\20.5.45456.45889\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\onprem\20.5.45456.45889\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\onprem\20.5.45456.45889\platform\WindowsPowerShellScripts\WebSearch
Copying Web Client Files
c:\dl\onprem\20.5.45456.45889\platform\WebClient\Microsoft Dynamics NAV
Copying Client Files
c:\dl\onprem\20.5.45456.45889\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\onprem\20.5.45456.45889\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\onprem\20.5.45456.45889\platform\LegacyDlls\systemFolder
Copying ModernDev Files
c:\dl\onprem\20.5.45456.45889\platform
c:\dl\onprem\20.5.45456.45889\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\onprem\20.5.45456.45889\platform\ConfigurationPackages
Copying Test Assemblies
C:\dl\onprem\20.5.45456.45889\platform\Test Assemblies
Copying Applications
C:\dl\onprem\20.5.45456.45889\au\Applications
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Restoring CRONUS Demo Database
Setting CompatibilityLevel for CRONUS on localhost\SQLEXPRESS
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Starting Business Central Service Tier
Importing license file
Your program license has expired.
at <ScriptBlock>, C:\Run\navinstall.ps1: line 331
at <ScriptBlock>, C:\Run\start.ps1: line 242
at <ScriptBlock>, <No file>: line 1Error
Using artifactUrl https://bcartifacts.azureedge.net/onprem/20.5.45456.45889/au
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\onprem\20.5.45456.45889\platform\ServiceTier\Program Files
c:\dl\onprem\20.5.45456.45889\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\onprem\20.5.45456.45889\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\onprem\20.5.45456.45889\platform\WindowsPowerShellScripts\WebSearch
Copying Web Client Files
c:\dl\onprem\20.5.45456.45889\platform\WebClient\Microsoft Dynamics NAV
Copying Client Files
c:\dl\onprem\20.5.45456.45889\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\onprem\20.5.45456.45889\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\onprem\20.5.45456.45889\platform\LegacyDlls\systemFolder
Copying ModernDev Files
c:\dl\onprem\20.5.45456.45889\platform
c:\dl\onprem\20.5.45456.45889\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\onprem\20.5.45456.45889\platform\ConfigurationPackages
Copying Test Assemblies
C:\dl\onprem\20.5.45456.45889\platform\Test Assemblies
Copying Applications
C:\dl\onprem\20.5.45456.45889\au\Applications
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Restoring CRONUS Demo Database
Setting CompatibilityLevel for CRONUS on localhost\SQLEXPRESS
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Starting Business Central Service Tier
Importing license file
Your program license has expired.
at <ScriptBlock>, C:\Run\navinstall.ps1: line 331
at <ScriptBlock>, C:\Run\start.ps1: line 242
at <ScriptBlock>, <No file>: line 1
New-BcContainer Telemetry Correlation Id: 26fd7789-377f-4c89-88ee-da27a967b9c2
Initialization of container BC205 failed
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\4.0.14\ContainerHandling\Wait-NavContainerReady.ps1:42 char:17
+ ...             throw "Initialization of container $containerName failed" ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Initialization of container BC205 failed:String) [], RuntimeException
    + FullyQualifiedErrorId : Initialization of container BC205 failed
```

**Comment by freddydk** [comment]
This seems to be an error on the "DVD"
When I download the AU version from here https://www.microsoft.com/en-us/download/details.aspx?id=104514
and look in the database folder - it does have a license file, which expires this year.
I cannot really change the published artifacts, but I can (and have earlier) patched them on the fly.
I have added a patch in Download-Artifacts to patch the license file for this version, it will look like this during download:
![image](https://user-images.githubusercontent.com/10775043/233976792-2fd9b8b7-2dee-47ad-b002-885217f46120.png)

**Comment by freddydk** [comment]
Shipped in 5.0.0
